flow/raster_cache.cc

```diff
@@ -419,37 +400,85 @@ void RasterCache::PrepareNewFrame() {
   display_list_cached_this_frame_ = 0;
 }
 
+void RasterCache::SweepOneCacheAfterFrame(RasterCacheKey::Map<Entry>& cache,
+                                          RasterCacheMetrics& picture_metrics,
+                                          RasterCacheMetrics& layer_metrics) {
+  std::vector<RasterCacheKey::Map<Entry>::iterator> dead;
+
+  for (auto it = cache.begin(); it != cache.end(); ++it) {
+    RasterCacheKeyType type = it->first.type();
```

Comment:
Perhaps add a `kind()` method  to the RasterCacheKey.

A follow-on task would be to develop a kLayerChildren facility that would cache the layer children rather than the layer itself. This would eliminate the need for the GetCacheableChild() mechanism used by OpacityLayer, ImageFilteredLayer, etc. At that point, the test for which kind of cache metrics to produce would involve a number of tests for kPic/kDL vs kLayer/kLChildren and it would be nice to have that logic centralized.

(This reminds me that I should add that task to the RasterCache umbrella issue)

Reply:
(Issue filed)

Reply:
Done

Reply:
My apologies for not being clear, I was looking for a second classification. type would still be the type of object being cached, but this second classification would describe the metrics type it is accounted under. By having this on the key we avoid a lot of duplicate "type == kPicture || type == kDisplaylist" tests throughout the metrics gathering code - that code would instead test for "kind == kPictureMetrics" instead and not have to know the details about which "type"s are accounted under which metrics.

Reply:
Done, thanks!